### PR TITLE
show tooltips for func overlay without transition

### PR DESCRIPTION
**Pull request by mmomtchev**
#5621 

How about this? The user will have to add the classes himself when he is using a function `overlay`

### CODE REVIEW

**Comment by kyletsang** [comment]
I think this will need to target the bs5 branch since it's breaking behavior.

**Comment by jquense** [comment]
I'm tempted to call this a bug fix...do you think anyone would be depending on the tooltip not working without a transition? I guess someone might be using it to toggle visibility?

**Comment by kyletsang** [comment]
I was initially thinking people were using the `show` prop that's passed along to toggle visibility manually, but maybe a lot of people just use transitions by default.  This sounds pretty low risk, so I'm down for merging this.

**Comment by jquense** [comment]
We can always revert if it causes a problem

**Review by jquense — APPROVED** [review]

File: src/Overlay.tsx [hunk 1]
```diff
@@ -205,6 +205,7 @@ function Overlay({
             ...overlayProps,
             placement,
             show,
+            className: !transition && show && 'show',
```
Comment (kyletsang) [inline src/Overlay.tsx:208]:
```suggestion
            ...(!transition && show && { className: 'show' }),
```

This change will avoid overwriting `className` when transition is true and spreading props after a className is declared.

**Comment by mmomtchev** [comment]
@jquense Not that many in fact :smile: 
The main problem is of course that recent versions of `create-react-app` have strict mode enabled - but no matter one's goodwill - and I tried hard - you end up disabling it for any serious project

File: src/Overlay.tsx (hunk 1 above)
Comment (mmomtchev) [reply src/Overlay.tsx:208]: `overlayProps` won't contain the `className` if `overlay` is a function? The `className` will probably be hard-coded in the function.

Comment (kyletsang) [reply src/Overlay.tsx:208]: There's no `className` in the `overlayProps`.  This is what's passed back from react-overlays:

https://github.com/react-bootstrap/react-overlays/blob/f9f1a14752a8d8ff9aa7c7d207aa0c413ca70660/src/Overlay.tsx#L117-L130

The above suggestion will prevent the below `className="abc"` from being overwritten when `transition=true`.
```
(props) => <Tooltip className="abc" {...props}>my tooltip</Tooltip
```

**Comment by kyletsang** [comment]
Thanks!

**Pull request merged**